hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -471,6 +471,11 @@ object DataSourceWriteOptions {
     .key("hoodie.deltastreamer.source.kafka.value.deserializer.schema")
     .noDefaultValue()
     .withDocumentation("")
+
+  val HIVE_BATCH_SYNC_PARTITION_NUM: ConfigProperty[String] = ConfigProperty
```

Comment:
please move this config under `HIVE_SYNC_AS_DATA_SOURCE_TABLE` to make hive_sync config clustered.